src/absil/ilwrite.fs

```diff
@@ -1494,7 +1500,10 @@ let GetCallsigAsStandAloneSigIdx cenv env info =
 let EmitLocalSig cenv env (bb: ByteBuffer) (locals: ILLocals) = 
     bb.EmitByte e_IMAGE_CEE_CS_CALLCONV_LOCAL_SIG
     bb.EmitZ32 locals.Length
-    locals |> ILList.iter (fun l -> EmitType cenv env bb l.Type) 
+    locals |> ILList.iter (fun l -> 
+        if l.IsPinned then 
+            bb.EmitByte et_PINNED
+        EmitType cenv env bb l.Type) 
```

Comment:
Can we please remove the duplication for this logic?
How about a new function `EmitTypeOfLocal`?